n3fit/src/n3fit/hyper_optimization/hyper_scan.py

```diff
@@ -109,7 +111,15 @@ def hyper_scan_wrapper(replica_path_set, model_trainer, hyperscanner, max_evals=
     # Tell the trainer we are doing hpyeropt
     model_trainer.set_hyperopt(True, keys=hyperscanner.hyper_keys, status_ok=hyperopt.STATUS_OK)
     # Generate the trials object
-    trials = filetrials.FileTrials(replica_path_set, parameters=hyperscanner.as_dict())
+    trials = FileTrials(replica_path_set, parameters=hyperscanner.as_dict())
+    # Initialize seed for hyperopt
+    trials.rstate = np.random.default_rng(42)
```

Comment:
Should this always be the same instead of setting it from the runcard and/or depending on the replica number? I think this way we try the same setups for each replica (though randomization in the NN may still result in different performances and thus change the setups `hypeorpt` will try). 

If you want it to be fixed, please put any user defined constants at the top of the module. E.g. `HYPEROPT_SEED = 42` and 
```suggestion
    trials.rstate = np.random.default_rng(HYPEROPT_SEED)
```

Reply:
I suppose you were maybe going to change this anyway since you pointed out some troubles with restarting, but the whole `rstate` implementation that passes it through `FileTrials` seems very convoluted for something that is harcoded and only used in a single place. As it is now everything could be replaced by doing `trials=np.random.default_rng(42)` inside the `fmin` function.

Reply:
> Should this always be the same instead of setting it from the runcard and/or depending on the replica number? I think this way we try the same setups for each replica (though randomization in the NN may still result in different performances and thus change the setups `hypeorpt` will try).
> 
> If you want it to be fixed, please put any user defined constants at the top of the module. E.g. `HYPEROPT_SEED = 42` and

Thanks @RoyStegeman. Done. Indeed, this seed could be later input via runcard and we are making it constant for the moment (unless you say otherwise). The idea here is indeed warrant that the hyperopt keep tracks and show same hyperparameter history in direct and restart runs.

Reply:
> I suppose you were maybe going to change this anyway since you pointed out some troubles with restarting, but the whole `rstate` implementation that passes it through `FileTrials` seems very convoluted for something that is harcoded and only used in a single place. As it is now everything could be replaced by doing `trials=np.random.default_rng(42)` inside the `fmin` function.

`trials.rstate=np.random.default_rng(42)` is set when I am just starting a fresh hyperoptimizarion. This just sets a common starting point for all runs done afresh. Once your run is done or was interrupted, `hyperopt` will save the state of the `FileTrials` for every completed trial in the `tries.pkl` pickle file, including the last used `rstate` as attribute.

Now when I want to restart:
```
    # For restarts, reset the state of `FileTrials` saved in the pickle file
    if hyperscanner.restart_hyperopt:
        pickle_file_to_load = f"{replica_path_set}/tries.pkl"
        log.info("Restarting hyperopt run using the pickle file %s", pickle_file_to_load)
        trials = FileTrials.from_pkl(pickle_file_to_load)
```
`trials` now will contain the history of the previous hyperopt as well as implicitly (via its attribute `trial.rstate`) the `rstate` of the last completed trial. **Note** that now (when restarting) the initial `trial.rstate` is not equal to `np.random.default_rng(42)`.  I suspect that this was your doubt; in summary the whole process then do not just resume in just setting  `trials.rstate=np.random.default_rng(42)`.

Reply:
You're right, apologies, I wasn't paying enough attention.